src/libs/ViewportDetector.js

```diff
@@ -58,8 +58,8 @@ ViewportDetector.prototype._detectViewport = function () {
   }
   const rect = this._element.getBoundingClientRect();
 
-  // Detect Screen Bottom                           // Detect Screen Top
-  if (rect.top < innerHeight + this._margins.top && rect.bottom > 0 - this._margins.bottom) {
+// Detect Screen Bottom                           // Detect Screen Top                         //Detect Screen Right
+if (rect.top < innerHeight + this._margins.top && rect.bottom > 0 - this._margins.bottom && rect.right <= (window.innerWidth || document.documentElement.clientWidth)) {
```

Comment:
what if the page is horizontally scrolled?  Should we check `rect.left` and `rect.right` against `window.pageXOffset` and `window.pageXOffset + window.innerWidth`?

Reply:
I have updated this to check whether page is horizontally scrolled and item is fully in view. However, I noticed that if the steam carousel on mobile for news.yahoo.com is scrolled - it doesn't trigger ViewportDetector again. So the beacon is never fired if user scrolls within the carousel. It is called if the page is scrolled. So I think this needs to be fixed but haven't figured out how to do this yet.

Reply:
correction: pageXOffset is not needed, because BoundingClientRect() returns left/right values [relative to viewport coordinate](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) already.

The correct logic for checking left/right should be something like this (similar to the top/bottom check), which should cover both partial in viewport and fully in viewport cases:
```
// viewportWidth = window.innerWidth || document.documentElement.clientWidth;
rect.left < viewportWidth && rect.right > 0
```

Reply:
> However, I noticed that if the steam carousel on mobile for news.yahoo.com is scrolled - it doesn't trigger ViewportDetector again.

That's a separate issue.  Lets discuss offline first.